waku/protocol/v2/waku_store.nim

```diff
@@ -336,3 +336,38 @@ proc query*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc) {.asyn
     return
 
   handler(response.value.response)
+
+# TODO Move to better place
+type AccountUpdateFunc = proc(peerId: PeerId, amount: int) {.gcsafe.}
+
+# NOTE: Experimental, maybe incorporate as part of query call
+proc queryWithAccounting*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc,
```

Comment:
Reason for this change is:

1) Not disrupt public API (yet)
2) We need access to node, and peer. Callback only gives us messages, and inside function doesn't have a reference to node. With this handler we can callback to node object and then pass onto accounting module.

Reply:
Makes sense.
Should we consider collating all handlers into a single `WakuStoreListener` object and make that a parameter for the API call so we can more easily extend in future? Handlers inside the listener are optional and only triggered if set.

Reply:
Yeah was thinking if an open options map or similar would make sense here, in general.

@decanus IIRC you were doing something similar somewhere but forgot where exactly?

Reply:
Yeah, thats what the subscriptions are for. Or at least it sounds like it.

Reply:
So how would this work with subscribe? Do you get access to the same data with peer etc? Not obvious to me (from phone).

Reply:
Oh actually, I don't think these parts work, let me think and I'll get back to you.

Reply:
In any case, above is out of scope for initial PoC.